### PR TITLE
Revise ssl versions and ciphers hardening

### PR DESCRIPTION
**Pull request by voltone**

### CODE REVIEW

File: docs/secure_coding_and_deployment_hardening/ssl.md [hunk 1]
```diff
@@ -94,41 +93,96 @@ However, please note that the `ssl_crl_cache` module does not actually cache the
...
-    {mac, fun
-        (sha) -> false;
-        (_) -> true
-    end}]).
+
+PreferredCiphers = [
+  %% Cipher suites (TLS 1.3): TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
+  #{cipher => aes_128_gcm, key_exchange => any, mac => aead, prf => sha256},
+  #{cipher => aes_256_gcm, key_exchange => any, mac => aead, prf => sha384},
+  #{cipher => chacha20_poly1305, key_exchange => any, mac => aead, prf => sha256},
```
Comment (max-au) [inline docs/secure_coding_and_deployment_hardening/ssl.md:107]: I believe this requires reasonably modern OTP version (24?).

Should it also be mentioned in the guide? I suspect OTP 22 simply won't accept the format.

**Comment by voltone** [comment]
Right, the sample configuration was only tested with OTP 24. I did check the out-of-the-box score of prior versions.

I'll see if it is feasible to achieve the same configuration with `ssl:cipher_suites/2` and `ssl:filter_cipher_suites/2` on older versions. If not, I'll just add a note that this is 24 only. Or I can add an example with TLS 1.2 only for older releases, which should be easier: I wouldn't really recommend 1.3 on older OTP versions anyway.

**Comment by voltone** [comment]
I just tested it, going back as far as 22.2, and it seems to work just fine.

**Comment by max-au** [comment]
Given that we recommend OTP 22 as the lowest version, it makes sense.

**Pull request merged**